src/CDF/CDFObject.php

```diff
@@ -235,6 +235,9 @@ public function getAttributes() {
 
   /**
    * {@inheritdoc}
+   *
+   * @return \Acquia\ContentHubClient\CDFAttribute|void
```

Comment:
When I'm checking the return value of the variable, in this case it would be NULL isn't it? In this case do you think it would be more appropriate to change the return type to `\Acquia\ContentHubClient\CDFAttribute|null` and specify `?CDFAttribute` as the return data type for the function?

Reply:
`  /**
   * {@inheritdoc}
   *
   * @return \Acquia\ContentHubClient\CDFAttribute|void
   *   Attribute object.
   */
  public function getAttribute($id) {
    if (!empty($this->attributes[$id])) {
      return $this->attributes[$id];
    }
  }`

If the`!empty($this->attributes[$id])` condition is not satisfied then function will not return anything. Hence `void` is added.

Reply:
If we abide by the clean code rules, this function either should return something or nothing. There should not be in-between. If the function doesn't return an object it should return null.
```php
$value = $this->getAttribute("some_attribute");
var_dump($value);
```
The above code has two possible return types. Either a CDFAttribute object or, because a variable must have a value, NULL.